compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -2619,13 +2619,19 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
             case sym if sym.isType => sym.asType
           }.toList
 
-        def memberType(name: String): Symbol = typeMember(name)
-        def typeMember(name: String): Symbol =
+        def memberType(name: String): Symbol =
           self.unforcedDecls.find(sym => sym.name == name.toTypeName)
+        def typeMember(name: String): Symbol =
+          lookupPrefix.allMembers.iterator.map(_.symbol).collect {
+            case sym if sym.isType && sym.name.toString == name => sym.asType
+          }.toList.headOption.getOrElse(dotc.core.Symbols.NoSymbol)
```

Comment:
Traversing all members to find one with a specific name is really inefficient. Could you instead do:
```suggestion
          lookupPrefix.memberBasedOnFlags(name.toTypeName)
```

Reply:
updated